modules/react/popup/lib/hooks/useReturnFocus.tsx

```diff
@@ -53,19 +54,18 @@ export const useReturnFocus = createElemPropsHook(usePopupModel)(model => {
     }
   }, []);
 
-  const onKeyUp = React.useCallback((event: KeyboardEvent) => {
-    requiresFocusChangeRef.current = false;
-  }, []);
-
   // track mousedown events to determine if the mouse target is a focusable element. If it is, we
   // should not return focus
-  const onMouseDown = React.useCallback((event: MouseEvent) => {
-    elementRef.current = event.target as Element;
-  }, []);
-
-  const onMouseUp = React.useCallback(() => {
-    elementRef.current = null;
-  }, []);
+  const onMouseDown = React.useCallback(
+    (event: MouseEvent) => {
+      if (model.state.stackRef.current && event.target instanceof HTMLElement) {
+        if (!PopupStack.contains(model.state.stackRef.current, event.target)) {
+          elementRef.current = event.target;
+        }
```

Comment:
The top `if` statement makes Typescript happy. The second `if` statement makes sure we're only counting `mousedown` events that are outside the popup when tracking the next element the user is clicking on for focus element calculations.

Before we were using `mouseup` to null out the `elementRef`. This worked for elements inside the popup because the popup didn't close and the mouseup was fired. When clicking outside the popup, focus changes as part of the `mousedown` event and the `mouseup` event is unsubscribed. The `elementRef` was nulled out by the `useLayoutEffect` cleanup function.

The old logic worked fine with Cypress and React 17 because the `useLayoutEffect` callback was called synchronously BEFORE the `mousedown` event released the event loop. React 18 calls the `useLayoutEffect` callback asynchronously. Cypress calls a `mousedown`, `mouseup`, and `click` within the same frame of the event loop which is impossible for a user to do. That means the `mouseup` event was clearing the `elementRef` in Cypress and React 18.

This new logic doesn't rely on the `mouseup` quirk that fails Cypress and React 18 and instead does popup contains detection manually and clears the `elementRef` consistently when the `useLayoutEffect` callback fires.